manifests/addon.pp

```diff
@@ -1,72 +1,115 @@
+# @summary Defined type for deploying Splunk Add-ons and Apps from either OS packages or via splunkbase compatible archives
 #
-# Defined type: splunk::addon
-#
-# This define sets up a TA (Technology Addon) for Splunk.  It (optionally)
-# installed a package, and configures input forwarders in
-# $SPLUNK_HOME/etc/apps/<app name>/local/inputs.conf
+# @example Basic usage
+#   splunk::addon { 'Splunk_TA_nix':
+#     splunkbase_source => 'puppet:///modules/splunk_qd/addons/splunk-add-on-for-unix-and-linux_602.tgz',
+#     inputs            => {
+#       'monitor:///var/log'       => {
+#        'whitelist' => '(\.log|log$|messages|secure|auth|mesg$|cron$|acpid$|\.out)',
+#        'blacklist' => '(lastlog|anaconda\.syslog)',
+#        'disabled'  => 'false'
+#       },
+#       'script://./bin/uptime.sh' =>  {
+#         'disabled' => 'false',
+#         'interval' => '86400',
+#         'source' => 'Unix:Uptime',
+#         'sourcetype' => 'Unix:Uptime'
+#       }
+#     }
+#   }
 #
-# Examples
+# @see https://docs.splunk.com/Documentation/AddOns/released/Overview/AboutSplunkadd-ons
 #
-# splunk::addon { 'search':
-#   package_manage => false,
-# }
+# @param splunk_home
+#   Overrides the default Splunk installation target values from Class[splunk::params]
 #
-# splunk::addon::input { 'monitor:///var/log/messages':
-#   attributes => {
-#     'index' => 'server_t',
-#   },
-# }
+# @param package_manage
+#   If a package should be installed as part of declaring a new instance of Splunk::Addon
 #
-# Alternatively you can feed inputs directly into splunk::addon using the
-# inputs parameter (useful if you are configuring from Hiera)
+# @param splunkbase_source
+#   When set the add-on will be installed from a splunkbase compatible archive instead of OS packages
 #
+# @param package_name
+#  The OS package to install if you are not installing via splunk compatible archive
 #
-# splunk::addon { 'search':
-#   package_manage => false,
-#   inputs          => {
-#     'monitor:///var/log/messages' => {
-#       'attributes' => {
-#         'index' => 'server_t',
-#       }
-#     }
-#   }
-# }
+# @param inputs
+#  A hash of inputs to be configured as part of add-on installation, alterntively you can also define splunk_input or splunkforwarder_input resouces seperately
 #
 define splunk::addon (
   Optional[Stdlib::Absolutepath] $splunk_home = undef,
   Boolean $package_manage                     = true,
+  Optional[String[1]] $splunkbase_source      = undef,
   Optional[String[1]] $package_name           = undef,
   Hash $inputs                                = {},
 ) {
 
-  include 'splunk::params'
+  include splunk::params
+
+  if defined(Class['splunk::forwarder']) {
+    $mode = 'forwarder'
+  } else {
```

Comment:
Is it worth checking that `Class['splunk']` has been defined?

Reply:
Class['splunk'] is now completely empty since the 7.2 refactor: https://github.com/voxpupuli/puppet-splunk/pull/215/files#diff-60ae41fd0a31977447947f59940ee9a4

Reply:
Sorry, `splunk::enterprise` then.

Reply:
Wouldn't hurt.